crates/node/core/src/args/log.rs

```diff
@@ -110,8 +113,78 @@ impl LogArgs {
             tracer = tracer.with_file(file, info);
         }
 
-        let guard = tracer.init()?;
-        Ok(guard)
+        #[cfg(feature = "opentelemetry")]
+        if self.otel.url.is_some() {
+            if let Ok(otel) = (&self.otel).try_into() {
+                tracer = tracer.with_otlp(otel);
+            } else {
+                tracing::warn!("Failed to parse otel config, skipping otel layer");
+            }
+        }
+
+        tracer
+    }
+
+    /// Initializes tracing with the configured options from cli args.
+    ///
+    /// Returns the file worker guard, and the file name, if a file worker was configured.
+    pub fn init_tracing(&self) -> eyre::Result<Option<FileWorkerGuard>> {
+        self.tracer().init()
+    }
+}
+
+/// Arguments for the `OpenTelemetry` tracing layer.
+#[cfg(feature = "opentelemetry")]
+#[derive(Debug, Args)]
+pub struct OtelArgs {
+    /// Endpoint url to which to send spans and events. The endpoint URL must
+    /// be a valid URL, including the protocol prefix (http or https) and any
+    /// http basic auth information.
+    ///
+    /// If the endpoint is not specified, events and spans will not be exported
+    /// to any external service.
+    ///
+    /// Available only when compiled with the `opentelemetry` feature.
+    #[arg(long = "otel.endpoint", value_name = "ENDPOINT", global = true)]
+    pub url: Option<String>,
+
+    /// The protocol to use for sending spans and events. Values are "grpc",
+    /// "binary", or "json".
+    ///
+    /// Available only when compiled with the `opentelemetry` feature.
+    #[arg(long = "otel.protocol", value_name = "PROTOCOL", global = true, default_value = "json")]
+    pub protocol: reth_tracing::OtlpProtocols,
+
+    /// The log level to use for the `OpenTelemetry` layer. Values are
+    /// "trace", "debug", "info", "warn", and "error". The default is "info".
+    /// Events and spans above this level will not be exported. Events and spans
+    /// at this level or below will be exported. See the documentation of
+    /// [`Level`] for more information.
+    ///
+    /// To specify `off`, omit the `--otel.url` argument.
+    ///
+    /// Available only when compiled with the `opentelemetry` feature.
+    #[arg(long = "otel.level", value_name = "LEVEL", global = true, default_value = "info")]
+    pub level: tracing::Level,
+
+    /// The timeout for sending spans and events, in milliseconds.
+    ///
+    /// Available only when compiled with the `opentelemetry` feature.
+    #[arg(long = "otel.timeout", value_name = "TIMEOUT", global = true, default_value = "1000")]
+    pub timeout: u64,
+}
+
+#[cfg(feature = "opentelemetry")]
+impl TryFrom<&OtelArgs> for reth_tracing::OtlpConfig {
```

Comment:
not a fan of tryfrom ref impls, can we convert this to a function of OtelArgs instead?

Reply:
yep, done